iceoryx2-cli/iox2/src/commands.rs

```diff
@@ -10,221 +10,270 @@
 //
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
+use anyhow::{anyhow, Context, Result};
+use cargo_metadata::MetadataCommand;
 use colored::*;
 use std::env;
-use std::ffi::OsStr;
 use std::fs;
-use std::io;
+use std::path::Path;
 use std::path::PathBuf;
 use std::process::{Command, Stdio};
-use thiserror::Error;
 
 #[derive(Clone, Debug, PartialEq)]
-enum CommandType {
+pub enum CommandType {
     Installed,
     Development,
 }
 
 #[derive(Clone, Debug)]
-struct CommandInfo {
-    path: PathBuf,
-    command_type: CommandType,
+pub struct CommandInfo {
+    pub name: String,
+    pub path: PathBuf,
 }
 
-#[derive(Error, Debug)]
-pub enum ExecutionError {
-    #[error("Command not found: {0}")]
-    NotFound(String),
-    #[error("Execution failed: {0}")]
-    Failed(String),
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PathsList {
+    build: Vec<PathBuf>,
+    install: Vec<PathBuf>,
 }
 
-pub fn list() {
-    println!("{}", "Installed Commands:".bright_green().bold());
+#[cfg(windows)]
+const PATH_SEPARATOR: char = ';';
```

Comment:
Just path separator can be ambiguous since it is also used for `/` vs `\`.

What do you think of `PATH_ENV_VAR_SEPARATOR`?